### PR TITLE
fix: chart import validation

### PR DESCRIPTION
**Pull request by dpgaspar**
### SUMMARY
Similar to https://github.com/apache/superset/pull/26887

Adds a missing validation to import charts when a low privilege user imports a dashboard that already exists.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: tests/unit_tests/charts/commands/importers/v1/import_test.py [hunk 1]
```diff
@@ -17,104 +17,130 @@
...
+    )
+    session.add(slice)
+    session.flush()
+
+    yield session
+    session.rollback()
+
+
+@pytest.fixture
+def session_with_schema(session: Session) -> Generator[Session, None, None]:
```
Comment (betodealmeida) [inline tests/unit_tests/charts/commands/importers/v1/import_test.py:65]: Nice!

**Review by betodealmeida — APPROVED** [review]
This is awesome! And thanks for fixing this!

File: superset/commands/chart/importers/v1/utils.py [hunk 2]
```diff
@@ -55,6 +55,12 @@ def import_chart(
     can_write = ignore_permissions or security_manager.can_access("can_write", "Chart")
     existing = session.query(Slice).filter_by(uuid=config["uuid"]).first()
     if existing:
+        if overwrite and can_write and hasattr(g, "user") and g.user:
```
Comment (john-bodley) [inline superset/commands/chart/importers/v1/utils.py:58]: I thought we had a shorthand method for checking whether the user was set, i.e., `get_username()` or similar.

Comment (dpgaspar) [reply superset/commands/chart/importers/v1/utils.py]: we don't exactly, I've created a new one

**Pull request merged**